pkg/controller/scheduler_hooks.go

```diff
@@ -23,17 +23,29 @@ import (
 	"github.com/fluxcd/flagger/pkg/canary"
 )
 
-func (c *Controller) runConfirmTrafficIncreaseHooks(canary *flaggerv1.Canary) bool {
+func (c *Controller) runConfirmTrafficIncreaseHooks(canary *flaggerv1.Canary, canaryController canary.Controller) bool {
 	for _, webhook := range canary.GetAnalysis().Webhooks {
 		if webhook.Type == flaggerv1.ConfirmTrafficIncreaseHook {
 			err := CallWebhook(canary.Name, canary.Namespace, flaggerv1.CanaryPhaseProgressing, webhook)
```

Comment:
I think this'd be better, as the webhook payload should have the current phase, which could be `WaitingTrafficIncrease` as well and not always `Progressing`.

```suggestion
			err := CallWebhook(canary.Name, canary.Namespace, canary.Status.Phase, webhook)
```


I see that `runConfirmRolloutHooks()` below also calls `CallWebhok()` with `CanaryPhaseProgressing` and not `cd.Status.Phase`. @stefanprodan could you confirm whether this is a bug or it's intentional and I'm missing something here?

Reply:
I applied this suggestion, and rebased + force pushed